torchtitan/config_manager.py

```diff
@@ -480,6 +480,20 @@ def __init__(self):
             help="Flight recorder ring buffer size, >0 means recording by default, 0 means disabled",
         )
 
+        # estimation mode settings
+        self.parser.add_argument(
+            "--estimate.memory",
+            help="Whether to estimate memory usage for FSDP",
+            default=False,
+        )
+
+        self.parser.add_argument(
+            "--estimate.mode",
+            type=str,
+            default="fake",
+            help="Mode of estimation to use ['fake', 'real']",
```

Comment:
What's the difference between these two? Would you please update the help message to be more informative to users?

From what I see, we just need to differentiate between `fake` and `real` (e.g. to determine whether or not to use the `FakeTensorMode` context manager). If that's the case, IMO it's better to have a boolean config, e.g. `--memory_estimation.fake_mode_only = true` by default.

Reply:
Yes I think this make sense. I will update this.